src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs

```diff
@@ -100,7 +116,7 @@ impl MinOverlappingPicker {
                     .then_with(|| (x.1 - x.0).cmp(&(y.1 - y.0)))
             })
             .unwrap();
-        let select_input_ssts = select_tables[*left..(right + 1)].to_vec();
+        let select_input_ssts = select_tables[*left..(*right + 1)].to_vec();
         let target_input_ssts = self
```

Comment:
Could we record the `target_file_range` when calculating `scores` and reuse the range instead of `check_base_level_overlap`?

Reply:
Because I do not want store too much in memory....

Reply:
I refactor it to avoid store statistic in memory